### PR TITLE
docs(copilot): clarify subagent execution model and orchestration patterns

### PR DESCRIPTION
**Pull request by digitarald**
Addresses user confusion about subagent behavior. Clarifies synchronous execution, parallel spawning, UI visibility, agent-initiated invocation, custom agent model overrides, agents array overriding disable-model-invocation, and adds orchestration patterns (coordinator/worker, multi-perspective code review). Replaces #9330 (was accidentally targeted at vnext).

### CODE REVIEW

File: docs/copilot/agents/subagents.md [hunk 1]
```diff
@@ -106,9 +132,24 @@ Consolidate findings into a single review summary.
...
 
-Hint in your chat prompt that you want to use a subagent to perform a specific task. The main agent will start a subagent, pass the task to it, and receive only the final result.
+### Agent-initiated vs user-invoked
+
+Subagents are typically **agent-initiated**, not directly invoked by users in chat. The main agent decides when context isolation helps—you don't manually type "run a subagent" for every task.
+
+The pattern works like this:
+
+1. You (or your custom agent's instructions) describe a complex task.
+2. The main agent recognizes that part of the task benefits from isolated context.
```
Comment (ntrogh) [inline docs/copilot/agents/subagents.md:142]:
```suggestion
2. The main agent recognizes the part of the task that benefits from isolated context.
```

File: docs/copilot/agents/subagents.md [hunk 2]
```diff
@@ -17,9 +17,35 @@ Keywords:
 
 Subagents in Visual Studio Code provide context isolation and enable you to run tasks in a dedicated context window, separate from the main agent session. This allows you to delegate complex or multi-step tasks to autonomous subagents without affecting the context window of the main chat session and helps it stay focused on the primary task. VS Code can run multiple subagents in parallel to improve overall performance.
 
-By default, subagents use the same agent and model as the main chat session. By running a custom agent as a subagent, you can apply specialized behavior, tools, and models for specific tasks. For example, use a research custom agent as a subagent to gather information and perform research tasks.
+By default, subagents use the same model and tools as the main chat session but start with a clean context window. Subagents do not inherit the main agent's instructions or conversation history—they receive only the task prompt you provide. By running a custom agent as a subagent, you can apply specialized behavior, tools, and models for specific tasks. For example, use a research custom agent as a subagent to gather information and perform research tasks.
```
Comment (ntrogh) [inline docs/copilot/agents/subagents.md:20]:
```suggestion
By default, subagents use the same model and tools as the main chat session but start with a clean context window. Subagents do not inherit the main agent's instructions or conversation history, they receive only the task prompt you provide. By running a custom agent as a subagent, you can apply specialized behavior, tools, and models for specific tasks. For example, use a research custom agent as a subagent to gather information and perform research tasks.
```

File: docs/copilot/agents/subagents.md [hunk 3]
```diff
@@ -17,9 +17,35 @@ Keywords:
...
+flowchart LR
+    A["Main Agent"] -->|"Spawns with task prompt"| B["Subagent\n(clean context)"]
+    B -->|"Works autonomously"| C["Research, analyze,\nexplore"]
+    C -->|"Returns summary"| A
+    A -->|"Continues with result"| D["Next step"]
+```
+
+## How subagent execution works
+
+Subagents are **synchronous**—the main agent waits for subagent results before continuing. This blocking behavior is intentional: subagent findings typically inform the next step of the task. Without the subagent results, the main agent lacks the information it needs to proceed effectively.
```
Comment (ntrogh) [inline docs/copilot/agents/subagents.md:32]:
```suggestion
Subagents are **synchronous**: the main agent waits for subagent results before continuing. This blocking behavior is intentional: subagent findings typically inform the next step of the task. Without the subagent results, the main agent lacks the information it needs to proceed effectively.
```

File: docs/copilot/agents/subagents.md [hunk 4]
```diff
@@ -17,9 +17,35 @@ Keywords:
...
+
+However, VS Code can spawn **multiple subagents in parallel**. When you request parallel analysis (for example, "analyze security, performance, and accessibility simultaneously"), VS Code runs those subagents concurrently and waits for all results before the main agent continues.
+
+> [!NOTE]
+> Subagents are different from starting a new agent session. A new session creates an entirely separate conversation with no connection to your current task. Subagents maintain the relationship—they do focused work and report back to the main agent, which stays in control of the overall task.
```
Comment (ntrogh) [inline docs/copilot/agents/subagents.md:37]:
```suggestion
> Subagents are different from starting a new agent session. A new session creates an entirely separate conversation with no connection to your current task. Subagents maintain the relationship: they do focused work and report back to the main agent, which stays in control of the overall task.
```

File: docs/copilot/agents/subagents.md [hunk 5]
```diff
@@ -106,9 +132,24 @@ Consolidate findings into a single review summary.
 
 To invoke a subagent in chat, the `runSubagent` tool must be enabled. This tool allows the main agent to delegate tasks to a subagent that operates in an isolated context window.
 
-Hint in your chat prompt that you want to use a subagent to perform a specific task. The main agent will start a subagent, pass the task to it, and receive only the final result.
+### Agent-initiated vs user-invoked
+
+Subagents are typically **agent-initiated**, not directly invoked by users in chat. The main agent decides when context isolation helps—you don't manually type "run a subagent" for every task.
```
Comment (ntrogh) [inline docs/copilot/agents/subagents.md:137]:
```suggestion
Subagents are typically **agent-initiated**, not directly invoked by users in chat. The main agent decides when context isolation helps. You don't manually need to type "run a subagent" for every task.
```

File: docs/copilot/agents/subagents.md [hunk 6]
```diff
@@ -127,7 +168,7 @@ Then update the docs/ folder with the new documentation.
 
 ## Run a custom agent as a subagent (Experimental)
 
-By default, a subagent inherits the agent from the main chat session and uses the same model and tools. To define specific behavior for a subagent, use a [custom agent](/docs/copilot/customization/custom-agents.md).
+By default, a subagent inherits the agent from the main chat session and uses the same model and tools. To define specific behavior for a subagent, use a [custom agent](/docs/copilot/customization/custom-agents.md). Custom agents can specify their own model, tools, and instructions—when used as a subagent, these settings override the defaults inherited from the main session.
```
Comment (ntrogh) [inline docs/copilot/agents/subagents.md:171]:
```suggestion
By default, a subagent inherits the agent from the main chat session and uses the same model and tools. To define specific behavior for a subagent, use a [custom agent](/docs/copilot/customization/custom-agents.md). Custom agents can specify their own model, tools, and instructions. When used as a subagent, these settings override the defaults inherited from the main session.
```

File: docs/copilot/agents/subagents.md [hunk 7]
```diff
@@ -181,6 +225,88 @@ Implement the following feature using test-driven development. Use subagents to
 3. Use the Refactor agent to improve the code quality
 ```
 
+## Orchestration patterns
+
+Subagents enable **orchestration patterns** where a coordinator agent delegates work to specialized worker agents. This approach helps you build sophisticated workflows while keeping each agent focused on what it does best.
+
+### Coordinator and worker pattern
+
+A coordinator agent manages the overall task and delegates subtasks to specialized subagents. Each worker agent can have a tailored set of tools—for example, planning and review agents need only read-only access, while the implementer needs edit capabilities.
```
Comment (ntrogh) [inline docs/copilot/agents/subagents.md:234]:
```suggestion
A coordinator agent manages the overall task and delegates subtasks to specialized subagents. Each worker agent can have a tailored set of tools. For example, planning and review agents need only read-only access, while the implementer needs edit capabilities.
```

File: docs/copilot/agents/subagents.md [hunk 8]
```diff
@@ -181,6 +225,88 @@ Implement the following feature using test-driven development. Use subagents to
...
+When asked to review code, run these subagents in parallel:
+- Correctness reviewer: logic errors, edge cases, type issues.
+- Code quality reviewer: readability, naming, duplication.
+- Security reviewer: input validation, injection risks, data exposure.
+- Architecture reviewer: does this fit the codebase patterns?
+
+After all subagents complete, synthesize findings into a prioritized summary. Note which issues are critical versus nice-to-have. Acknowledge what the code does well.
+```
+
+This pattern works because each subagent approaches the code fresh, without being anchored by what other perspectives found. In this example, the orchestrator shapes each subagent's focus area through its prompt—a lightweight approach that requires no additional agent files.
```
Comment (ntrogh) [inline docs/copilot/agents/subagents.md:305]:
```suggestion
This pattern works because each subagent approaches the code fresh, without being anchored by what other perspectives found. In this example, the orchestrator shapes each subagent's focus area through its prompt. This is a lightweight approach that requires no additional agent files.
```

File: docs/copilot/agents/subagents.md [hunk 9]
```diff
@@ -181,6 +225,88 @@ Implement the following feature using test-driven development. Use subagents to
...
+
+> [!TIP]
+> For more control, each review perspective can be its own custom agent with specialized tool access. For example, a security reviewer might use a security-focused MCP server, while a code quality reviewer might have access to linting CLI tools. This lets each perspective use the best tools for its specific focus.
```
Comment (ntrogh) [inline docs/copilot/agents/subagents.md:308]:
```suggestion
> For more control, each review perspective can be its own custom agent with specialized tool access. For example, a security reviewer might use a security-focused MCP server, while a code-quality reviewer might have access to linting CLI tools. This lets each perspective use the best tools for its specific focus.
```

File: docs/copilot/agents/subagents.md [hunk 10]
```diff
@@ -17,9 +17,35 @@ Keywords:
...
 
-By default, subagents use the same agent and model as the main chat session. By running a custom agent as a subagent, you can apply specialized behavior, tools, and models for specific tasks. For example, use a research custom agent as a subagent to gather information and perform research tasks.
+By default, subagents use the same model and tools as the main chat session but start with a clean context window. Subagents do not inherit the main agent's instructions or conversation history—they receive only the task prompt you provide. By running a custom agent as a subagent, you can apply specialized behavior, tools, and models for specific tasks. For example, use a research custom agent as a subagent to gather information and perform research tasks.
+
+```mermaid
+flowchart LR
+    A["Main Agent"] -->|"Spawns with task prompt"| B["Subagent\n(clean context)"]
```
Comment (ntrogh) [inline docs/copilot/agents/subagents.md:24]: I think the Mermaid renderer needs `<br>` instead of newline characters.
Also, currently we don't support Mermaid on our website - can you create a PNG and use that instead? You can put the Mermaid in comments to have the source info.

**Review by ntrogh — CHANGES_REQUESTED** [review]
@digitarald Great additions to the doc!! I left some minor comments (a.o. I don't like em-dashes ;) )

**Review by xztmhs888 — APPROVED** [review]

**Review by xztmhs888 — APPROVED** [review]
